Source/Row/Binding.swift

```diff
@@ -347,10 +347,7 @@ extension Data: Binding {
 /// The `CodableBinding` protocol represents a `Binding` that is also `Codable`.
 public protocol CodableBinding: Codable, Binding {}
 
-extension CodableBinding {
-    /// The binding type of a parameter to bind to a statement.
-    public typealias BindingType = Data
-
+extension CodableBinding where BindingType == Data {
```

Comment:
This is the change worth calling out. The Swift 4.1 compiler throws a warning now when defining the typealias in the protocol or the protocol extension. And rightfully so, because it can lead to all sorts of odd behavior when redefining the `typealias` in conforming types or subclasses.

Reply:
Read the thread and SR and the change seems like the right thing to do, but to make sure I'm grokking this... with the new syntax, this extension is only available when the `BindingType` type is `Data`, correct?

How does this break existing users of the code?

Reply:
Great question. The only place this would break an existing user is if you had created a `CodableBinding` type such as a `Person` struct that you wanted to write directly into the database as a data blob. We actually have these in the test suite.

Reply:
Ah, makes sense. I'd suggest #1 as the version strategy. While this does require changes to the callers code, it's more of a slight language shift caused by Swift 4.1 than a "we actually changed the API" thing. This doesn't feel like a major version change to me.